File: Examples/VisualBasic/AsposePDF/Bookmarks/UpdateBookmarks.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf
Namespace AsposePDF.Bookmarks
    Public Class UpdateBookmarks
        Public Shared Sub Run()
            ' ExStart:UpdateBookmarks
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Bookmarks()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("UpdateBookmarks.pdf"))

            ' Get a bookmark object
            Dim pdfOutline As OutlineItemCollection = pdfDocument.Outlines(1)
            pdfOutline.Title = "Updated Outline"
            pdfOutline.Italic = True
            pdfOutline.Bold = True

            dataDir = dataDir & Convert.ToString("UpdateBookmarks_out_.pdf")
            ' Save output
            pdfDocument.Save(dataDir)
            ' ExEnd:UpdateBookmarks
            Console.WriteLine(Convert.ToString(vbLf & "Bookmarks updated successfully." & vbLf & "File saved at ") & dataDir)

        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePDF/Bookmarks/UpdateBookmarks.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;

namespace CSharp.AsposePdf.Bookmarks
{
    public class UpdateBookmarks
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Bookmarks();

            //open document
            Document pdfDocument = new Document(dataDir + "UpdateBookmarks.pdf");

            //get a bookmark object
            OutlineItemCollection pdfOutline = pdfDocument.Outlines[1];
            pdfOutline.Title = "Updated Outline";
            pdfOutline.Italic = true;
            pdfOutline.Bold = true;
                        
            //save output
            pdfDocument.Save(dataDir + "UpdateBookmarks_out.pdf");

        }
    }
}